src/main/java/net/sf/jabref/JabRefPreferences.java

```diff
@@ -1337,6 +1368,26 @@ public ExternalFileType getExternalFileTypeByMimeType(String mimeType) {
     }
 
     /**
+     * Look up the external file type name registered for this MIME type, if any.
+     *
+     * @param mimeType The MIME type.
+     * @return The name of the ExternalFileType registered, or null if none. For the mime type "text/html", a valid file type name is
+     *         guaranteed to be returned.
+     */
+    public String getExternalFileTypeNameByMimeType(String mimeType) {
+        for (ExternalFileType type : externalFileTypes) {
+            if ((type.getMimeType() != null) && type.getMimeType().equals(mimeType)) {
+                return type.getName();
+            }
+        }
+        if ("text/html".equals(mimeType)) {
+            return HTML_FALLBACK_TYPE.getName();
+        } else {
+            return null;
+        }
+    }
```

Comment:
From a refactoring point of view, I would suggest to create a class which contains all the external file type methods as JabRefPreferences should manage the preferences, but not external file types.